cyverse_allocation/cyverse_rules_engine_setup.py

```diff
@@ -80,79 +77,144 @@ def cannot_renew_allocation_source(self):
+                "operator": "is_true",
+                "value": True
+            }
+        )
+        return conditions
+
+    # Does the Allocation Source renewal strategy match `strategy_name`?
+    conditions.append(
+        {
+            "name": "renewal_strategy",
+            "operator": "equal_to",
+            "value": strategy_name
+        }
+    )
+
+    # Is the Allocation Source valid? (Basically valid if it has not been end-dated.)
+    conditions.append(
+        {
+            "name": "is_valid",
+            "operator": "is_true",
+            "value": True
+        }
+    )
+
+    assert period_type in ['days', 'on_calendar_day']
+
+    if period_type == 'on_calendar_day':
+        calendar_day_to_renew_on = period_param
+        assert type(calendar_day_to_renew_on) == int, 'Invalid calendar day: Must be an integer'
+        assert 0 < calendar_day_to_renew_on <= 31, 'Invalid calendar day: Must be an integer from 1 to 31'
+        # TODO: What happens when the cron job doesn't fire on the first of the month?
+        # Add a condition to check if it's been more than a month since last renewal?
+
+        # Make sure it's been more than a day since we last renewed
+        conditions.append(
+            {
+                "name": "days_since_renewed",
+                "operator": "greater_than_or_equal_to",
+                "value": 1
+            }
+        )
+        conditions.append(
+            {
+                "name": "today_calendar_day",
+                "operator": "equal_to",
+                "value": calendar_day_to_renew_on
+            }
+        )
+
+    if period_type == 'days':
+        renewed_in_days = period_param
+        assert type(renewed_in_days) == int, 'Invalid number of days: Must be an integer'
+        conditions.append(
+            {
+                "name": "days_since_renewed",
+                "operator": "greater_than_or_equal_to",
+                "value": renewed_in_days
+            }
+        )
 
     return conditions
 
 
-def _create_actions_for(compute_allowed, renewed_in_days):
+def _create_actions_for(strategy_name, strategy_config):
     actions = []
+    compute_allowed = strategy_config.get('compute_allowed', 0)
+    period_type = strategy_config.get('period_type')
 
-    if renewed_in_days > 0:
-        actions.append({"name": "renew_allocation_source",
-                        "params": {"compute_allowed": compute_allowed}
-                        })
-
-    else:
+    if period_type is None:
         actions.append({"name": "cannot_renew_allocation_source"})
+        return actions
+
+    actions.append(
+        {
+            "name": "renew_allocation_source",
+            "params":
+                {
+                    "strategy_name": strategy_name,
+                    "compute_allowed": compute_allowed
+                }
+        }
+    )
 
     return actions
 
 
 # RENEWAL STRATEGY CONFIGURATION
 renewal_strategies = {
```

Comment:
it might be helpful to include some treatment of "key is `strategy_name` for the `renewal_startegies` might help to convey the data-definition. 

Not necessary - but for an outlander like me, it made that "jump" and I think I'm right. Just would be cool to _know_, :100:.